src/main/java/life/qbic/portal/portlet/StatusPortlet.java

```diff
@@ -91,227 +53,231 @@
+        .setRenderer(new DateRenderer(new SimpleDateFormat("dd.MM.yyyy '/' HH:mm")));
+    logTable.addColumn("Place", String.class);
+    logTable.addColumn("Status", String.class);
+    logTable.addColumn("Responsible Person", String.class);
+    logTable.addColumn("Contact", String.class);
 
-        //panelContent.addComponent(inputLayout);
-        panelContent.addComponent(logTable);
+    ////////////////////////////////////////////////////
 
+    panelContent.addComponent(idField);
+    panelContent.addComponent(trackButton);
 
-        queryPanel.setContent(panelContent);
+    // panelContent.addComponent(inputLayout);
+    panelContent.addComponent(logTable);
 
-        trackButton.addClickListener(new Button.ClickListener() {
-            public void buttonClick(Button.ClickEvent event) {
-                //Notification.show("trying API");
-                //testAPIRequest();
 
-                logTable.getContainerDataSource().removeAllItems();
+    queryPanel.setContent(panelContent);
 
-                ////////////////////////////////////////////
+    trackButton.addClickListener(new Button.ClickListener() {
+      public void buttonClick(Button.ClickEvent event) {
+        // Notification.show("trying API");
+        // testAPIRequest();
 
-                if (idField.getValue().equals("")){
+        logTable.getContainerDataSource().removeAllItems();
 
-                    logTable.getContainerDataSource().removeAllItems();
-                }else {
+        ////////////////////////////////////////////
 
-                    try {
+        String sampleCode = idField.getValue();
+        if (sampleCode != null && !sampleCode.isEmpty()) {
 
-                        //String baseURL = "http://services.qbic.uni-tuebingen.de:8080/sampletrackingservice/";
-                        if(serviceList.size() == 0){
-                            throw new Exception("No available backend services");
-                        }
+          try {
 
-                        String baseURL = serviceList.get(0).getRootUrl().toString() + "/";
+            if (serviceList.size() == 0) {
+              throw new Exception("No available backend services");
+            }
 
-                        ////////////////////////////
-                        //auth
+            String baseURL = serviceList.get(0).getRootUrl().toString() + "/";
 
-                        byte[] credentials = Base64.encodeBase64((httpUser.name + ":" + httpUser.password).getBytes(StandardCharsets.UTF_8));
-                        String authHeader = "Basic " + new String(credentials, StandardCharsets.UTF_8);
+            // test if user is allowed to see sample
+            List<ch.systemsx.cisd.openbis.generic.shared.api.v1.dto.Sample> samples =
+                openbis.getParentsBySearchService(sampleCode);
+            if (samples.isEmpty() || !spaces.contains(samples.get(0).getSpaceCode())) {
+              LOG.error("User tried searching for " + sampleCode
```

Comment:
I would log the two occurences individually, as the second one is more severe, that the first one.

Reply:
And maybe make a small helper method `userIsAuthorized(sample)`, to make it more declarative? Just a suggestion

Reply:
good suggestion, fixed